src/main/java/org/spongepowered/api/block/entity/carrier/chest/Chest.java

```diff
@@ -64,7 +64,7 @@ public interface Chest extends NameableCarrierBlockEntity {
      * @return The attachment type of this chest.
      */
     default Value.Mutable<ChestAttachmentType> attachmentType() {
-        return this.requireValue(Keys.CHEST_ATTACHMENT_TYPE).asMutable();
+        return this.block().requireValue(Keys.CHEST_ATTACHMENT_TYPE).asMutable();
```

Comment:
I know this is subtle, but why?

Reply:
Because `Keys.CHEST_ATTACHMENT_TYPE` is registered for `BlockState`, not `BlockEntity`. And currently this method simply throw an exception

Reply:
Note that we have code (`SpongeDataHolder#impl$delegateDataHolder`) that could make this work on BlockEntity too.
e.g. 
https://github.com/SpongePowered/Sponge/blob/api-12/src/main/java/org/spongepowered/common/world/server/SpongeServerLocation.java#L280

Reply:
Reverted change in `Chest` file, in impl made `BlockEntity` delegate `DataHolder` to its `#block()` and `#block().type().
Also reverteв `Container#type()` change since it's breaking (and in api-12 Faithcaio fixed this).